pkg/local_object_storage/writecache/persist.go

```diff
@@ -46,24 +46,37 @@ func (c *cache) persistLoop() {
 
 func (c *cache) persistToCache(objs []objectInfo) []int {
 	var (
-		failMem []int
+		failMem []int // some index is negative => all objects starting from it will overflow the cache
 		doneMem []int
 	)
 	var sz uint64
 	err := c.db.Update(func(tx *bbolt.Tx) error {
 		b := tx.Bucket(defaultBucket)
+		cacheSz := c.estimateCacheSize()
 		for i := range objs {
 			if uint64(len(objs[i].data)) >= c.smallObjectSize {
 				failMem = append(failMem, i)
 				continue
 			}
 
+			// check if object will overflow write-cache size limit
+			updCacheSz := c.incSizeDB(cacheSz)
+			if updCacheSz > c.maxCacheSize {
+				// set negative index
+				failMem = append(failMem, -i)
```

Comment:
Are there possible issues with `-0` index?

Reply:
Sure, we should handle zero case.